tests/nn/misc/test_checkpoint_activations.py

```diff
@@ -3,70 +3,103 @@
+"""Test fairscale.nn.misc.checkpoint_activations API."""
 
 import unittest
 
 import torch
 import torch.nn as nn
-from torch.utils.checkpoint import checkpoint
+from torch.utils.checkpoint import checkpoint as torch_checkpoint
 
 from fairscale.nn.misc.checkpoint_activations import checkpoint_wrapper
 
 
+def get_cuda_mem_allocated():
+    """Helper to get cuda memory allocated if possible."""
+    if torch.cuda.is_available():
+        return torch.cuda.memory_allocated()
+    else:
+        return 0
+
+
 class Model(nn.Module):
-    def __init__(self, use_pytorch_checkpoint=False, use_fairseq_checkpoint=False, **kwargs):
+    def __init__(self, use_pytorch_checkpoint=False, use_fairscale_checkpoint=False, **kwargs):
         super().__init__()
-        torch.manual_seed(0)
+        torch.manual_seed(0)  # make sure weights are deterministic.
+        assert not (
+            use_pytorch_checkpoint and use_fairscale_checkpoint
+        ), "Cannot use both pytorch and fairscale checkpointing mechanisms."
         self.use_pytorch_checkpoint = use_pytorch_checkpoint
         self.ffn = nn.Sequential(
             nn.Linear(32, 128),
             # add a Dropout layer to test RNG save/restore
             nn.Dropout(p=0.5),
             nn.Linear(128, 32),
         )
-        if use_fairseq_checkpoint:
+        if use_fairscale_checkpoint:
             self.ffn = checkpoint_wrapper(self.ffn, **kwargs)
         self.out = nn.Linear(32, 1)
 
     def forward(self, x):
         if self.use_pytorch_checkpoint:
-            x = checkpoint(self.ffn, x)
+            x = torch_checkpoint(self.ffn, x)
         else:
             x = self.ffn(x)
         return self.out(x)
 
 
 class TestComparisonToPyTorch(unittest.TestCase):
-    def _test_checkpoint_wrapper(self, device, log_memory_usage=False):
-        def get_loss_and_gnorm(model):
-            torch.manual_seed(1)
-            input = torch.rand(2, 16, 32).requires_grad_(True).to(device)
+    def _test_checkpoint_wrapper(self, device):
+        def get_loss_and_gnorm(model, input):
+            ret = {}
+            ret["mem_0"] = get_cuda_mem_allocated()
             model.zero_grad()
             loss = model(input).sum()
+            ret["mem_after_fwd"] = get_cuda_mem_allocated()
             loss.backward()
+            ret["mem_after_bwd"] = get_cuda_mem_allocated()
             gnorm = torch.norm(torch.stack([torch.norm(p.grad.detach()) for p in model.parameters()]))
-            return {"loss": loss, "gnorm": gnorm}
+            ret["loss"] = loss.item()
+            ret["gnorm"] = gnorm.item()
+            return ret
 
+        input = torch.rand(2, 16, 32).requires_grad_(True)
         model = Model().to(device)
-        no_cpt = get_loss_and_gnorm(model)
+        no_cpt = get_loss_and_gnorm(model, input.to(device))
 
         model = Model(use_pytorch_checkpoint=True).to(device)
-        pyt_cpt = get_loss_and_gnorm(model)
+        pyt_cpt = get_loss_and_gnorm(model, input.to(device))
+
```

Comment:
it's important to reset memory counters, call gc, etc. between calls, otherwise it's not measuring memory properly.

Try this pattern:
```python
def reset_everything():
    torch.cuda.synchronize()
    gc.collect()
    torch.cuda.empty_cache()
    torch.cuda.reset_peak_memory_stats()
```

and also use `torch.cuda.max_memory_allocated()` to get the peak memory usage over a window